--- a/lib/portage/_sets/dbapi.py	2022-01-23 20:34:14.047039118 -0300
+++ b/lib/portage/_sets/dbapi.py	2022-01-23 20:38:03.467311839 -0300
@@ -105,6 +105,8 @@
         if not exclude_paths:
             for link, p in vardb._owners.iter_owners(paths):
                 pkg = pkg_str(link.mycpv, None)
+                if "sys-kernel/debian" in pkg.cp:
+                    continue
                 rValue.add("%s:%s" % (pkg.cp, pkg.slot))
         else:
             all_paths = set()
@@ -114,6 +116,8 @@
             for link, p in vardb._owners.iter_owners(all_paths):
                 pkg = pkg_str(link.mycpv, None)
                 atom = "%s:%s" % (pkg.cp, pkg.slot)
+                if "sys-kernel/debian" in pkg.cp:
+                    continue
                 rValue.add(atom)
                 # Returned paths are relative to ROOT and do not have
                 # a leading slash.
