--- a/bin/ebuild.sh	2022-01-23 20:22:44.350369907 -0300
+++ b/bin/ebuild.sh	2022-01-23 20:30:08.038739255 -0300
@@ -686,11 +686,11 @@
 if [[ $EBUILD_PHASE = depend ]] ; then
 	export SANDBOX_ON="0"
 	set -f
-
+	_PY_COMPAT="${PYTHON_COMPAT[@]}"
 	metadata_keys=(
 		DEPEND RDEPEND SLOT SRC_URI RESTRICT HOMEPAGE LICENSE
 		DESCRIPTION KEYWORDS INHERITED IUSE REQUIRED_USE PDEPEND BDEPEND
-		EAPI PROPERTIES DEFINED_PHASES IDEPEND INHERIT
+		EAPI PROPERTIES DEFINED_PHASES IDEPEND INHERIT _PY_COMPAT
 	)
 
 	if ! ___eapi_has_BDEPEND; then
@@ -707,6 +707,7 @@
 		echo "${f}=$(echo ${!f})" >&${PORTAGE_PIPE_FD} || exit $?
 	done
 	exec {PORTAGE_PIPE_FD}>&-
+	unset _PY_COMPAT
 	set +f
 else
 	# Note: readonly variables interfere with __preprocess_ebuild_env(), so
