--- a/lib/_emerge/EbuildPhase.py	2022-01-23 20:42:55.123649422 -0300
+++ b/lib/_emerge/EbuildPhase.py	2022-01-23 20:46:34.559897141 -0300
@@ -23,6 +23,7 @@
     _prepare_workdir,
     _prepare_fake_distdir,
     _prepare_fake_filesdir,
+    _prepare_fake_repodir,
 )
 from portage.util import writemsg, ensure_dirs
 from portage.util._async.AsyncTaskFuture import AsyncTaskFuture
@@ -246,6 +247,7 @@
             alist = self.settings.configdict["pkg"].get("A", "").split()
             _prepare_fake_distdir(self.settings, alist)
             _prepare_fake_filesdir(self.settings)
+            _prepare_fake_repodir(self.settings)
 
         fd_pipes = self.fd_pipes
         if fd_pipes is None:
--- a/lib/portage/package/ebuild/doebuild.py	2022-01-23 20:49:20.828081524 -0300
+++ b/lib/portage/package/ebuild/doebuild.py	2022-01-23 20:50:38.500166718 -0300
@@ -492,6 +492,7 @@
     mysettings["T"] = os.path.join(mysettings["PORTAGE_BUILDDIR"], "temp")
     mysettings["SANDBOX_LOG"] = os.path.join(mysettings["T"], "sandbox.log")
     mysettings["FILESDIR"] = os.path.join(settings["PORTAGE_BUILDDIR"], "files")
+    mysettings["REPODIR"] = os.path.join(settings["PORTAGE_BUILDDIR"], "repo")
 
     # Prefix forward compatability
     eprefix_lstrip = mysettings["EPREFIX"].lstrip(os.sep)
--- a/lib/portage/package/ebuild/prepare_build_dirs.py	2022-01-23 20:55:35.256486820 -0300
+++ b/lib/portage/package/ebuild/prepare_build_dirs.py	2022-01-23 20:55:15.792466076 -0300
@@ -499,6 +499,18 @@
             os.symlink(real_filesdir, symlink_path)
 
 
+def _prepare_fake_repodir(settings):
+    real_repodir = os.path.normpath(os.path.join(os.path.dirname(settings["EBUILD"]), "../.."))
+    symlink_path = os.path.dirname(settings["FILESDIR"]) + "/repo"
+    try:
+        link_target = os.readlink(symlink_path)
+    except OSError:
+        os.symlink(real_repodir, symlink_path)
+    else:
+        if link_target != real_repodir:
+            os.unlink(symlink_path)
+            os.symlink(real_repodir, symlink_path)
+
 def _prepare_fake_distdir(settings, alist):
     orig_distdir = settings["DISTDIR"]
     edpath = os.path.join(settings["PORTAGE_BUILDDIR"], "distdir")
